types/index.d.ts

```diff
@@ -262,7 +263,7 @@ export function renderHook<
   Container extends RendererableContainer | HydrateableContainer = HTMLElement,
   BaseElement extends RendererableContainer | HydrateableContainer = Container,
 >(
-  render: (initialProps: Props) => Result,
+  render: (initialProps: Props, initialArgs: Props) => Result,
```

Comment:
This looks wrong. It is now a function taking exactly 2 arguments, but should be a function taking a spread argument based on an array type.

I think to make this typesafe, we would need another options type, and an overloaded signature, something like:
```ts
export interface RenderHookArgsOptions<Args extends any[], ...> extends ... {
    initialArgs: Args
}
```
(with required `initialArgs` to prevent ambiguity), and an _additional_
```
export function renderHook<Result, Args extends any[], ...>(
  render(...initialArgs: Args) => Result,
  options: RenderHookArgsOptions<Args, ...>,
): RenderHookArgsResult<Result, Args>
```
and a new `RenderHookArgsResult<Result, Args>` containing a `rerender(...args: Args)` function.